### PR TITLE
feat: add react-talend-components support

### PR DESCRIPTION
**Pull request by jmfrancois**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [ ] Docs have been added / updated

**What is the current behavior?** (You can also link to an open issue here)

The set of icons can't be used by react-talend-components

**What is the new behavior?**

The set of Icons can be used by react-talend-components

**Other information**:

The main issue we have is the index.html is loading a set of string. We want our React <Icon>

### CODE REVIEW

**Comment by acateland** [comment]
So i have another question, this create a file with every icon saved in js variable, so react can inject it in the document.

So if we have 1000 occurence of one icon, it is present 1000 time in the DOM.

Shouldn't we require the generated svg spritesheet who descrive each icon as a symbol, and have the Icon component generate a svg document with a link to the symbol

```
<svg class="icon">
  <use xlink:href="#icon-id" />
</svg>
```
the avantage of this solution is that the DOM doesn't grow much since the svg elements of the icon are not repeated X time in the DOM while keeping the ability to style the elements by targeting the svg class icon in this case.

File: README.md [hunk 1]
```diff
@@ -2,6 +2,34 @@
 
 This is the set of SVG icons used in our apps.
 
+# How to use
+
+To use this icon set you just have to install it throw NPM.
```
Comment (jsomsanith) [inline README.md]: Typo here : throw --> through

File: README.md [hunk 2]
```diff
@@ -2,6 +2,34 @@
...
+
+```shell
+npm install talend-icons
+```
+
+Next you can get the icon set object throw a simple require:
```
Comment (jsomsanith) [inline README.md]: Same Typo here : throw --> through

File: README.md [hunk 3]
```diff
@@ -2,6 +2,34 @@
...
+
+```javascript
+var icons = require('talend-icons');
+icons.add.toSVG();
+<svg ...></svg>
+icons.add.toSVGuse();
+<svg ...><use xlink:href="#add" /></svg>
+```
+
+This icon set is registred within [react-talend-components](https://github.com/Talend/react-talend-components) Icon. So you can use it throw this addon.
```
Comment (jsomsanith) [inline README.md]: Same Typo here : throw --> through

**Comment by jsomsanith** [comment]
I approve since we don't have a lot of icons for now but I agree with @acateland

**Comment by jmfrancois** [comment]
yes @acateland it's an optimisation that can be done later.
the main idea is the following.
We will add a SVGProvider component in react-talend-component that will take values from Icon.register.

So the main idea could be the following. If SVGProvider is used the Icon rendering will use reference, else it will use the current implementation.

**Pull request merged**